ngraph/core/src/op/pad.cpp

```diff
@@ -200,16 +200,35 @@ bool op::v1::Pad::evaluate_pad(const HostTensorVector& outputs, const HostTensor
     } else {
         pad_value = pad_zero_value.data();
     }
+
+    // compute pads_begin and pads_end CoordinateDiffs from pads_begin 
```

Comment:
@ilyachur I guess the logic of `validate_and_infer_types()` should be revised due it relies on `get_pads_begin()` (and _end) which work correctly only for Constant ops (so I propose to remove or extend `get_pads_begin()` (and _end))